### PR TITLE
Port changes for installing GPU driver from 2022.3 branch

### PR DESCRIPTION
**Pull request by artanokhov**
### Details:
 - Removed install_NEO_OCL_driver.sh
 - Added ocl-icd lib for install_openvino_depend
 - Ported changes to configurations-for-intel-gpu.md

### Tickets:
 - 79326

### CODE REVIEW

**Review by kblaszczak-intel — APPROVED** [review]

File: scripts/install_dependencies/install_openvino_dependencies.sh [hunk 1]
```diff
@@ -163,6 +166,7 @@ elif [ "$os" == "centos7" ] || [ "$os" == "centos8" ] ||
     fi
 
     pkgs_dev=(gcc gcc-c++ make glibc libstdc++ libgcc cmake3 "json-devel.$arch" "zlib-devel.$arch" sudo)
+    pkgs_gpu=()
```
Comment (ilya-lavrenov) [inline scripts/install_dependencies/install_openvino_dependencies.sh:169]: why is it empty?
For Ubuntu we need `ocl-icd-libopencl1`, some packages are also available for other platforms

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 2]
```diff
@@ -7,39 +7,24 @@
 
 @endsphinxdirective
 
+In case if you are intended to use OpenVINO GPU plugin and offload network inference to Intel® graphics processor, the Intel Graphics Driver should be properly configured on your system.
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:10]:
```suggestion
If you are intended to use OpenVINO™ GPU plugin and offload inference to Intel® Processor Graphics (GPU), Intel® Graphics Driver must be properly configured on your system.
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 3]
```diff
@@ -7,39 +7,24 @@
...
 
+If it is already installed, and you want to keep it, you can skip the installation steps.
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:12]:
```suggestion
If you have already installed Intel® Graphics Driver and would like to keep it, you can skip the installation steps below.
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 4]
```diff
@@ -7,39 +7,24 @@
...
 
 ## Linux
 
-If you have installed OpenVINO Runtime from the archive file, APT, or YUM, follow these steps to work with GPU:
+To install the latest available **Intel® Graphics Compute Runtime for OpenCL™** for your OS, see the [Install Guides](https://github.com/intel/compute-runtime/releases/latest).
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:16]:
```suggestion
To install the latest available **Intel® Graphics Compute Runtime for OpenCL™** for your operating system, see the [Install Guides for Intel® Graphics Compute Runtime for oneAPI Level Zero and OpenCL™ Driver](https://github.com/intel/compute-runtime/releases/latest).
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 5]
```diff
@@ -7,39 +7,24 @@
...
 
-1. Go to the install_dependencies directory:
-   ```sh
-   cd <INSTALL_DIR>/install_dependencies/
-   ```
+> **NOTE**: If you use RedHat 8 OS please install OpenCL library as prerequisite via following command line:
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:18]:
```suggestion
> **NOTE**: If you are using RedHat 8, use the following command to install the OpenCL library as a prerequisite:  
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 6]
```diff
@@ -7,39 +7,24 @@
...
-   ```
+> **NOTE**: If you use RedHat 8 OS please install OpenCL library as prerequisite via following command line:  
+> ```sh rpm -ivh http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/ocl-icd-2.2.12-1.el8.x86_64.rpm```
 
-2. Install the **Intel® Graphics Compute Runtime for OpenCL™** driver components required to use the GPU plugin and write custom layers for Intel® Integrated Graphics. The drivers are not included in the package. To install it, run this script:
-   ```sh
-   sudo -E ./install_NEO_OCL_driver.sh
-   ```
-   > **NOTE**: If you installed OpenVINO Runtime via PyPI, you can get this script from [the OpenVINO repository](https://github.com/openvinotoolkit/openvino/blob/master/scripts/install_dependencies/install_NEO_OCL_driver.sh).
+> **NOTE**: For instructions specific to discrete graphics platforms, refer to [the dgpu guide](https://dgpu-docs.intel.com/installation-guides/index.html) (Intel® Arc™ A-Series Graphics, Intel® Data Center GPU Flex Series, Intel® Data Center GPU MAX Series, Intel® processor graphics Gen12, and Intel® Iris Xe MAX codename DG1).
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:21]:
```suggestion
For instructions for discrete graphics platforms, refer to [the dGPU guides](https://dgpu-docs.intel.com/installation-guides/index.html), including Intel® Arc™ A-Series Graphics, Intel® Data Center GPU Flex Series, Intel® Data Center GPU MAX Series, Intel® processor graphics Gen12, and Intel® Iris Xe MAX codename DG1.
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 7]
```diff
@@ -7,39 +7,24 @@
...
 
-   > **NOTE**: To use the **Intel® Iris® Xe MAX Graphics**, see the [Intel® Iris® Xe MAX Graphics with Linux*](https://dgpu-docs.intel.com/devices/iris-xe-max-graphics/index.html) page for driver installation instructions.
-   
-   The script compares the driver version on the system to the current version. If the driver version on the system is higher or equal to the current version, the script does 
-   not install a new driver. If the version of the driver is lower than the current version, the script uninstalls the lower version and installs the current version with your permission:
-   ![](../img/NEO_check_agreement.png) 
+You may consider installing one of the earlier versions of the driver, based on your particular setup needs.
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:23]:
```suggestion
You can also install one of the earlier versions of the driver based on your particular setup needs.
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 8]
```diff
@@ -7,39 +7,24 @@
...
 
-   Higher hardware versions require a higher driver version, namely 20.35 instead of 19.41. If the script fails to uninstall the driver, uninstall it manually. During the script execution, you may see the following command line output:  
-   ```sh
-   Add OpenCL user to video group    
-   ```
-   Ignore this suggestion and continue.<br>
-   You can also find the most recent version of the driver, installation procedure and other information on the [Intel® software for general purpose GPU capabilities](https://dgpu-docs.intel.com/index.html) site.
+It is recommended that you refer to the [Intel® Graphics Compute Runtime Github page](https://github.com/intel/compute-runtime/) for instructions and recommendations on GPU driver installation specific to particular releases, including the list of supported hardware platforms.
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:25]:
```suggestion
For instructions and recommendations on the installation of particular releases of the GPU driver, as well as the list of supported hardware platforms, see the [Intel® Graphics Compute Runtime GitHub page](https://github.com/intel/compute-runtime/).
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 9]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
 
 ## Windows
 
-This section will help you check if you require driver installation. Install indicated version or higher.
-
-If your applications offload computation to **Intel® Integrated Graphics**, you must have the Intel Graphics Driver for Windows installed on your hardware.
-[Download and install the recommended version](https://downloadcenter.intel.com/download/30079/Intel-Graphics-Windows-10-DCH-Drivers). 
+To install the Intel Graphics Driver for Windows on your hardware, please proceed with the [instruction](https://www.intel.com/content/www/us/en/support/articles/000005629/graphics.html).
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:36]:
```suggestion
To install the Intel® Graphics Driver for Windows on your hardware, proceed with [this article on the Support site](https://www.intel.com/content/www/us/en/support/articles/000005629/graphics.html). 
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 10]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
...
 
 To check if you have this driver installed:
 
 1. Type **device manager** in your **Search Windows** box and press Enter. The **Device Manager** opens.
-
-2. Click the drop-down arrow to view the **Display adapters**. You can see the adapter that is installed in your computer:
-   ![](../img/DeviceManager.PNG)
-
+2. Click the drop-down arrow to view the **Display adapters**. You can see the adapter that is installed in your computer:  
+![](../img/DeviceManager.PNG)
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:42]:
```suggestion
   ![](../img/DeviceManager.PNG)
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 11]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
...
 3. Right-click the adapter name and select **Properties**.
+4. Click the **Driver** tab to see the driver version.  
+![](../img/DeviceDriverVersion.PNG)
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:45]:
```suggestion
   ![](../img/DeviceDriverVersion.PNG)
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 12]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
...
-4. Click the **Driver** tab to see the driver version. 
-   ![](../img/DeviceDriverVersion.PNG)
 
 You are done updating your device driver and are ready to use your GPU.
+
+## Additional info
+
+In the internal OpenVINO validation the following versions of Intel Graphics Driver were used:
+
+Operation System | Driver version
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:54]:
```suggestion
Operating System | Driver version
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 13]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
...
+![](../img/DeviceDriverVersion.PNG)
 
-4. Click the **Driver** tab to see the driver version. 
-   ![](../img/DeviceDriverVersion.PNG)
 
 You are done updating your device driver and are ready to use your GPU.
+
+## Additional info
+
+In the internal OpenVINO validation the following versions of Intel Graphics Driver were used:
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:52]:
```suggestion
For your reference, the following versions of Intel® Graphics Driver are validated internally:
```

File: docs/install_guides/configurations-for-intel-gpu.md [hunk 14]
```diff
@@ -48,24 +33,31 @@ You've completed all required configuration steps to perform inference on proces
...
 3. Right-click the adapter name and select **Properties**.
+4. Click the **Driver** tab to see the driver version.  
+![](../img/DeviceDriverVersion.PNG)
 
-4. Click the **Driver** tab to see the driver version. 
-   ![](../img/DeviceDriverVersion.PNG)
 
 You are done updating your device driver and are ready to use your GPU.
+
+## Additional info
```
Comment (xu-yuan1) [inline docs/install_guides/configurations-for-intel-gpu.md:50]:
```suggestion
## Additional Information
```

File: scripts/install_dependencies/install_openvino_dependencies.sh (hunk 1 above)
Comment (artanokhov) [reply scripts/install_dependencies/install_openvino_dependencies.sh:169]: As I said in PR #15160 for ubu ocl-icd-libopencl1 listed as dependency by intel-opencl-icd

File: docs/install_guides/configurations-for-intel-gpu.md (hunk 2 above)
Comment (artanokhov) [reply docs/install_guides/configurations-for-intel-gpu.md:10]: This is port of changes from PR #15160 that's already approved by docs team.
Could you please clarify why we have such difference

**Review by evgenytalanin-intel — APPROVED** [review]

**Pull request merged**